File: app/src/main/java/com/tanacom/introrealm/java/CountryDataHelper.java
//package com.tanacom.introrealm.java;
//
//import com.tanacom.introrealm.kotlin.CountryModel;
//
//import io.realm.Realm;
//
//public class CountryDataHelper {
//
//    public static void addCountry(Realm realm, String countryName) {
//
//        realm.beginTransaction();
//
//        Long maxId = (Long) realm.where(CountryModel.class).max("id");
//
//        if (maxId == null) {
//            CountryModel model = realm.createObject(CountryModel.class, 0);
//            model.setName(countryName);
//        } else {
//            CountryModel model = realm.createObject(CountryModel.class, (maxId + 1));
//            model.setName(countryName);
//
//        }
//
//        realm.commitTransaction();
//
//
//    }
//
//    public static void deleteCountry(Realm realm, long id) {
//        realm.executeTransactionAsync(
//                new Realm.Transaction() {
//                    @Override
//                    public void execute(Realm realm) {
//                        CountryModel model = realm.where(CountryModel.class).equalTo("id", id).findFirst();
//                        if (model != null) {
//                            model.deleteFromRealm();
//                        }
//                    }
//                }
//        );
//    }
//
//
//
//}

File: app/src/main/java/com/tanacom/introrealm/kotlin/CountryDataHelper.kt
package com.tanacom.introrealm.kotlin

import io.realm.Realm

object CountryDataHelper {

    @JvmStatic
    fun addCountry(realm: Realm, countryName: String?) {


        realm.beginTransaction()

        val maxId = realm.where(CountryModel::class.java).max("id") as Long?

        if (maxId == null) {
            val model = realm.createObject(
                CountryModel::class.java, 0
            )
            model.name = countryName
        } else {
            val model = realm.createObject(
                CountryModel::class.java, maxId + 1
            )
            model.name = countryName
        }

        realm.commitTransaction()

    }

    fun syncCountry(realm: Realm, countryName: String?) {


        realm.beginTransaction()

        val maxId = realm.where(CountryModel::class.java).max("id") as Long?

        if (maxId == null) {
            val model = realm.createObject(
                CountryModel::class.java, 0
            )
            model.name = countryName
        } else {
            val model = realm.createObject(
                CountryModel::class.java, maxId + 1
            )
            model.name = countryName
        }

        realm.commitTransaction()

    }

    @JvmStatic
    fun deleteCountry(realm: Realm, id: Long) {
        realm.executeTransactionAsync { realm ->
            val model = realm.where(
                CountryModel::class.java
            ).equalTo("id", id).findFirst()
            model?.deleteFromRealm()
        }
    }


}